src/cpp/ext/csm/metadata_exchange.h

```diff
@@ -66,11 +68,65 @@ class ServiceMeshLabelsInjector : public LabelsInjector {
       absl::Span<const std::shared_ptr<std::map<std::string, std::string>>>
           optional_labels_span) const override;
 
+  const std::vector<std::pair<absl::string_view, std::string>>&
+  TestOnlyLocalLabels() const {
+    return local_labels_;
+  }
+
+  const grpc_core::Slice& TestOnlySerializedLabels() const {
+    return serialized_labels_to_send_;
+  }
+
  private:
   std::vector<std::pair<absl::string_view, std::string>> local_labels_;
   grpc_core::Slice serialized_labels_to_send_;
 };
 
+// A LabelsIterable class provided by ServiceMeshLabelsInjector. EXPOSED FOR
+// TESTING PURPOSES ONLY.
+class MeshLabelsIterable : public LabelsIterable {
+ public:
+  struct RemoteAttribute {
+    absl::string_view otel_attribute;
+    absl::string_view metadata_attribute;
+  };
+
+  enum class GcpResourceType : std::uint8_t { kGke, kGce, kUnknown };
+
+  MeshLabelsIterable(
+      const std::vector<std::pair<absl::string_view, std::string>>&
+          local_labels,
+      grpc_core::Slice remote_metadata);
+
+  absl::optional<std::pair<absl::string_view, absl::string_view>> Next()
+      override;
+
+  size_t Size() const override;
+
+  void ResetIteratorPosition() override { pos_ = 0; }
+
+  // Returns true if the peer sent a non-empty base64 encoded
+  // "x-envoy-peer-metadata" metadata.
+  bool GotRemoteLabels() const { return metadata_.struct_pb != nullptr; }
+
+ private:
+  struct StructPb {
```

Comment:
I don't think we need this struct anymore.  Instead, let's just put the two fields directly in the enclosing class.

Reply:
It is kind of convenient to use with `DecodeMetadata` so I can initialize the fields in the constructor itself.

Reply:
Removed `StructPb`